src/cpp/qos/ParameterList.cpp

```diff
@@ -21,26 +21,22 @@
 #include <fastrtps/qos/QosPolicies.h>
 
 #include <functional>
+#include <stdexcept>
 
 namespace eprosima {
 namespace fastrtps {
 
 using namespace rtps;
 
-#define IF_VALID_CALL() {                              \
-                            if(valid)                  \
-                            {                          \
-                                qos_size += plength;   \
-                                if(!processor(&p))     \
-                                    return false;      \
-                            }                          \
-                            else                       \
-                            {                          \
-                                return false;          \
-                            }                          \
-                            break;                     \
+#define IF_VALID_CALL() {                                        \
```

Comment:
Now that this macro is only used three times, it may be good to remove it completely. We can always do the `qos_size += plength;` after the switch, and do the processing of the data read inside an `if(valid)`.

Going even further, I see that we are always checking plength, so it is likely that `valid` will never become false

And even more, we may not need the temporary variables, and could directly access `msg->buffer[msg->pos + ...]`

Reply:
- qos_size handling has been moved outside of the switch
- parameter size checking has been moved to each of the params

This simplifies each case statement and the macro has been removed (not only on ParameterList, but also on the ProxyDatas)

Not so sure about the temporary variables: the classes already encapsulate the deserialization, and repeating the logic here seems redundant and less maintainable